### PR TITLE
core(fr): convert response-compression gatherer

### PR DESCRIPTION
**Pull request by adamraine**
Ref #11313

### CODE REVIEW

File: lighthouse-core/test/gather/gatherers/dobetterweb/response-compression-test.js [hunk 1]
```diff
@@ -7,194 +7,187 @@
 
 /* eslint-env jest */
 
+jest.mock('../../../../gather/driver/network.js', () => ({
+  fetchResponseBodyFromCache: jest.fn(),
```
Comment (adamraine) [inline lighthouse-core/test/gather/gatherers/dobetterweb/response-compression-test.js:11]: Perhaps use `mockDriverSubmodules` once https://github.com/GoogleChrome/lighthouse/pull/12478 lands

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 2]
```diff
@@ -30,9 +33,15 @@ const textResourceTypes = [
...
+class ResponseCompression extends FRGatherer {
+  /** @type {LH.Gatherer.GathererMeta<'DevtoolsLog'>} */
+  meta = {
+    supportedModes: ['navigation', 'navigation'],
+    dependencies: {DevtoolsLog: DevtoolsLog.symbol},
+  }
+
   /**
-   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @param {LH.Artifacts.NetworkRequest[]} networkRecords
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js:44]: hot takes over here ;)

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 3]
```diff
@@ -30,9 +33,15 @@ const textResourceTypes = [
   NetworkRequest.TYPES.EventSource,
 ];
 
-class ResponseCompression extends Gatherer {
+class ResponseCompression extends FRGatherer {
+  /** @type {LH.Gatherer.GathererMeta<'DevtoolsLog'>} */
+  meta = {
+    supportedModes: ['navigation', 'navigation'],
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js:39]:
```suggestion
    supportedModes: ['timespan', 'navigation'],
```

Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/dobetterweb/response-compression.js:39]: should probably add some config validation for that

File: lighthouse-core/test/gather/gatherers/dobetterweb/response-compression-test.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/test/gather/gatherers/dobetterweb/response-compression-test.js:11]: just landed it 👍

**Review by patrickhulce — COMMENTED** [review]
add it to the config as well?

**Comment by adamraine** [comment]
> add it to the config as well?

Yes indeed

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js (hunk 2 above)
Comment (adamraine) [reply lighthouse-core/gather/gatherers/dobetterweb/response-compression.js:44]: 😈

File: lighthouse-core/test/fraggle-rock/gather/mock-driver.js [hunk 4]
```diff
@@ -150,21 +150,24 @@ function createMockContext() {
 }
 
 function mockDriverSubmodules() {
-  const navigationMock = {gotoURL: jest.fn(),
-  };
```
Comment (patrickhulce) [inline lighthouse-core/test/fraggle-rock/gather/mock-driver.js:154]: haha woops, thanks for cleaning it up :)

**Review by patrickhulce — APPROVED** [review]
nice! LGTM

**Pull request merged**